src/nvim/eval.c

```diff
@@ -14266,7 +14266,8 @@ static void f_serverstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 
   int result = server_start((char *) rettv->vval.v_string);
   if (result != 0) {
-    EMSG2("Failed to start server: %s", uv_strerror(result));
+    EMSG2("Failed to start server: %s",
+          result > 0 ? "Unknonwn system error" : uv_strerror(result));
```

Comment:
Hm, `result` is here always `1`, so we could use the string "Unknown system error" unconditionally.

For a more specific error message we only have `errno`, but this variable is not always set on error in `server_start()`.